client/src/controllers/LoaderController.test.js

```diff
@@ -0,0 +1,99 @@
+import { Application } from '@hotwired/stimulus';
+import { LoaderController } from './LoaderController';
+
+describe('LoaderController', () => {
+  beforeEach(() => {
+    document.body.innerHTML = `
+            <button
+                type="submit"
+                class="button button-longrunning"
+                data-controller="w-loader"
+                data-w-loader-active-class="button-longrunning-active"
+                data-w-loader-active-value="Signing in…"
+                data-w-loader-duration-seconds-value="40"
+                data-action="w-loader#activate"
+            >
+                <svg>...</svg>
+                <em data-w-loader-target="label" id="em-el">Sign in</em>
+            </button>
+        `;
+    Application.start().register('w-loader', LoaderController);
+  });
+
+  afterEach(() => {
+    document.body.innerHTML = '';
+    jest.clearAllMocks();
+    jest.clearAllTimers();
+  });
+
+  it('should not execute the rest of the code if the form is not valid', () => {
```

Comment:
```suggestion
  it('should not change the button to loading if the form is not valid', () => {
```

Try to write your test descriptions in a way that reflects the behaviour not the logic.